File: edittextpicker/src/main/java/com/edittextpicker/aliazaz/TextUtils.java
package com.edittextpicker.aliazaz;

import android.text.InputFilter;

final class TextUtils {

    // call in afterTextChanged event
    public final static StringBuilder editTextLoopToNextChar(String maskEdit, int position) {

        StringBuilder finalResult = new StringBuilder();
        for (int i = position; i < maskEdit.length(); i++) {
            if (maskEdit.charAt(i) != '#') {
                finalResult.append(maskEdit.charAt(i));
            } else
                break;
        }

        return finalResult;
    }

    // set length
    public final static InputFilter[] setLengthEditText(String maskText) {
        return new InputFilter[]{new InputFilter.LengthFilter(maskText.length())};
    }

}

File: edittextpicker/src/main/java/com/edittextpicker/aliazaz/TextUtils.kt
package com.edittextpicker.aliazaz

import android.text.InputFilter
import android.text.InputFilter.LengthFilter

internal object TextUtils {

    // call in afterTextChanged event
    @JvmStatic
    fun editTextLoopToNextChar(maskEdit: String, position: Int): StringBuilder {
        val finalResult = StringBuilder()
        loop@ for (i in position until maskEdit.length) {
            /*if (maskEdit[i] != '#') {
                finalResult.append(maskEdit[i])
            } else break*/

            when {
                maskEdit[i] != '#' -> finalResult.append(maskEdit[i])
                else -> break@loop
            }
        }
        return finalResult
    }

    // set length
    @JvmStatic
    fun setLengthEditText(maskText: String): Array<InputFilter> = arrayOf(LengthFilter(maskText.length))

}